src/components/getting-started/fileblock.js

```diff
@@ -23,110 +23,87 @@ import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
 
-    copy(Helpers.dedent(this.props.children));
+  const copyCodeToClipboard = e => {
+    e.preventDefault();
 
-    this.setState({ clicked: false });
+    const contentToCopy = Helpers.dedent(children);
+    setCopyToClipboard(contentToCopy);
   };
 
-  classNames() {
-    var classNames = [];
+  const classNames = () => {
+    const classNames = ['codeblock--container'];
 
-    classNames.push('codeblock--container');
-    if (this.state.hovering) {
+    if (isHovering) {
       classNames.push('hovering');
     }
-    if (this.state.clicked) {
+    if (isCopiedToClipboard) {
       classNames.push('clicked');
     }
-    if (this.props.hideText) {
+    if (hideText) {
       classNames.push('oneline');
     }
 
     return classNames.join(' ');
-  }
+  };
 
-  blob() {
-    var blob = new Blob([Helpers.dedent(this.props.children)], {
+  const getFileAsBlob = () => {
+    const blob = new Blob([Helpers.dedent(children)], {
       type: 'application/plain;charset=utf-8',
     });
+
     return blob;
-  }
+  };
 
-  downloadAsFileLink() {
+  const downloadAsFileLink = () => {
     return (
-      <a
-        download={this.props.fileName}
-        href={window.URL.createObjectURL(this.blob())}
-      >
+      <a download={fileName} href={window.URL.createObjectURL(getFileAsBlob())}>
         <i aria-hidden='true' className='fa fa-file-download' />
       </a>
     );
-  }
+  };
 
-  render() {
-    return (
-      <div className={this.classNames()}>
-        <pre>
-          <div
-            className='content'
-            ref={d => {
-              this.pre = d;
-            }}
-          >
-            <div className='codeblock--filename'>{this.props.fileName}</div>
-            <div className='codeblock--filecontents'>
-              {this.props.hideText ? (
-                undefined
-              ) : (
-                <Line text={Helpers.dedent(this.props.children)} />
-              )}
-            </div>
-          </div>
-          <div
-            className='codeblock--buttons'
-            onMouseOut={function() {
-              this.setState({ hovering: false });
-            }.bind(this)}
-            onMouseOver={function() {
-              this.setState({ hovering: true });
-            }.bind(this)}
-          >
-            {Modernizr.adownload ? this.downloadAsFileLink() : null}
-            <a
-              href='#'
-              onClick={this.copyCodeToClipboard}
-              onMouseUp={function() {
-                this.setState({ clicked: true });
-              }.bind(this)}
-            >
-              <i aria-hidden='true' className='fa fa-content-copy' />
-            </a>
+  return (
+    <div className={classNames()}>
+      <pre>
+        <div className='content' ref={preElement}>
+          <div className='codeblock--filename'>{fileName}</div>
+          <div className='codeblock--filecontents'>
+            {hideText ? undefined : <Line text={Helpers.dedent(children)} />}
```

Comment:
We could do this kind of things this way instead, unless we need this `undefined`:

```javascript
{hideText && <Line text={Helpers.dedent(children)} />}
```